include/deal.II/base/utilities.h

```diff
@@ -549,23 +550,38 @@ namespace Utilities
    */
   template <typename T>
   size_t
-  pack(const T &          object,
-       std::vector<char> &dest_buffer,
-       const bool         allow_compression = true);
+  pack(
+    const T &          object,
+    std::vector<char> &dest_buffer,
+    const bool         allow_compression = true
+#ifdef DEAL_II_WITH_ZLIB
+    ,
+    const boost::iostreams::gzip_params &gzip_params =
+      boost::iostreams::gzip_params(boost::iostreams::gzip::default_compression)
+#endif
```

Comment:
That's generally making things difficult because you can't call the function with this argument depending on whether or not zlib was detected. A better approach is to allow the argument, but ignore it if zlib wasn't detected. Or does the `boost::iostream::gzip_params` class only exist in certain situations?

Reply:
Yes, if zlib is not available that class does not exist. I discussed with @marcfehling during the workshop that maybe we should drop the whole parameter. I have not seen significant differences in the size of compressed data, and we might just always use the default compression parameters. An alternative is to make this another bool parameter (use_best_compression). What do you think?

Reply:
Yes, I'd be ok with just dropping the argument and defaulting to the default :-)